celery.js

```diff
@@ -33,18 +33,19 @@ function Configuration(options) {
     self.broker_type = url.parse(self.BROKER_URL).protocol.slice(0, -1);
     debug('Broker type: ' + self.broker_type);
 
-    if (self.RESULT_BACKEND && self.RESULT_BACKEND.toLowerCase() === 'amqp') {
-        self.backend_type = 'amqp';
-    } else if (self.RESULT_BACKEND && url.parse(self.RESULT_BACKEND)
-        .protocol === 'redis:') {
+    if (self.RESULT_BACKEND && url.parse(self.RESULT_BACKEND).protocol === 'redis:') {
         self.backend_type = 'redis';
+    } else {
+        self.backend_type = 'amqp';
```

Comment:
Good question, we could check if it's a supported protocol otherwise throw an error maybe.